Explorer/Assets/DCL/Multiplayer/Chats/MultiplayerChatMessagesBus.cs

```diff
@@ -1,129 +1,78 @@
-        {
-            this.roomHub.IslandRoom().DataPipe.DataReceived -= DataPipeOnDataReceived;
-            this.roomHub.SceneRoom().DataPipe.DataReceived -= DataPipeOnDataReceived;
-        }
-
-        public event Action<ChatMessage>? OnMessageAdded;
-
-        public void Send(string message)
-        {
-            using var wrap = multiPool.TempResource<Packet>();
-            var packet = wrap.value;
-
-            using var chatWrap = multiPool.TempResource<Decentraland.Kernel.Comms.Rfc4.Chat>();
-            packet.Chat = chatWrap.value;
-
-            packet.Chat.Message = message;
-            packet.Chat.Timestamp = DateTime.UtcNow.TimeOfDay.TotalSeconds;
-
-            using var memoryWrap = memoryPool.Memory(packet);
-            packet.WriteTo(memoryWrap);
-
-            Send(memoryWrap.Span());
-        }
+            this.messageDeduplication = messageDeduplication;
 
-        private void Send(Span<byte> data)
-        {
-            Send(roomHub.IslandRoom(), data);
-        }
-
-        private static void Send(IRoom room, Span<byte> data)
-        {
-            room.DataPipe.PublishData(data, TOPIC, room.Participants.RemoteParticipantSids(), DataPacketKind.KindReliable);
+            messagePipesHub.IslandPipe().Subscribe<Decentraland.Kernel.Comms.Rfc4.Chat>(Packet.MessageOneofCase.Chat, OnMessageReceived);
+            messagePipesHub.ScenePipe().Subscribe<Decentraland.Kernel.Comms.Rfc4.Chat>(Packet.MessageOneofCase.Chat, OnMessageReceived);
         }
 
-        private void DataPipeOnDataReceived(ReadOnlySpan<byte> data, Participant participant, DataPacketKind kind)
+        private void OnMessageReceived(ReceivedMessage<Decentraland.Kernel.Comms.Rfc4.Chat> receivedMessage)
         {
-            if (TryParse(data, out Packet? response) == false)
-                return;
-
-            if (response!.MessageCase is Packet.MessageOneofCase.Chat)
-                HandleAsync(new SmartWrap<Packet>(response, multiPool), participant).Forget();
+            OnChatAsync(receivedMessage).Forget();
         }
 
-        private bool TryParse(ReadOnlySpan<byte> data, out Packet? packet)
+        private async UniTaskVoid OnChatAsync(ReceivedMessage<Decentraland.Kernel.Comms.Rfc4.Chat> receivedMessage)
         {
-            try
-            {
-                packet = packetParser.ParseFrom(data).EnsureNotNull();
-                return true;
-            }
-            catch (Exception e)
+            using (receivedMessage)
             {
-                ReportHub.LogWarning(
-                    ReportCategory.ARCHIPELAGO_REQUEST,
-                    $"Someone sent invalid packet: {data.Length} {data.HexReadableString()} {e}"
+                if (messageDeduplication.TryPass(receivedMessage.FromWalletId, receivedMessage.Payload.Timestamp) == false)
+                    return;
+
+                var profile = await profileRepository.GetAsync(receivedMessage.FromWalletId, 0, CancellationToken.None);
```

Comment:
It should contain a cancellation token: you can dispose of the `bus` from `DynamicWorldContainer.Dispose()`